sdjournal/read.go

```diff
@@ -188,6 +190,83 @@ process:
 	return
 }
 
+func (r *JournalReader) FollowWithFields(until <-chan time.Time, fields map[string]struct{}, writer io.Writer) (err error) {
```

Comment:
I like the idea of passing in the channel, because it allows me more flexibility when I write my programs with this library. For example, I am working on a daemon that forwards journald logs to an external logging service over http. The only reason I would ever want it killed is if a signal interrupt were given, so to do this I send a `time.Now()` from my signal handler into this channel to stop the program gracefully.